### PR TITLE
Hide title bar on mac, and change style with theme. Fixes Issue #1055

### PR DESCRIPTION
**Pull request by StormBurpee**
Hi, thought I'd take a go at this. 

Haven't got a windows machine near me to test it there, however you might want to look at making it a frameless window for windows. Then just add in custom close, max and minimise buttons. (Pretty easy, I can look at doing it for you if you like the look of this commit.) 
Also, if you went down this route, we could look at hiding the minimise and maximise buttons when the sidebar is folded (I think that's what you guys call it.), so it just shows the close button, and doesn't overlap. Again this is all personal preference, but it would look nice.

Had to do a few style changes with the modal for preferences, other things where just personal preference to make it align and look how I thought it looked best. However you might want to make some small changes to some of the positioning of the elements. (I'd probably move the preferences button into the top bar)

Other than that, this should achieve what is brought up in issue #1055 
The top bar is draggable and also changes with each style. :) Let me know what you guys think, or if you need me to change anything! It's pretty self explanatory however.

screen shot 2018-05-25 at 5 53 31 am
screen shot 2018-05-25 at 5 53 11 am
screen shot 2018-05-25 at 5 52 34 am
screen shot 2018-05-25 at 5 52 50 am

### CODE REVIEW

**Comment by ZeroX-DG** [comment]
@StormBurpee Wow, it look nice! But if you switch to the frameless window, how do you access the old window menu on top?
image

**Comment by StormBurpee** [comment]
Hi @ZeroX-DG I'll take a look at this now. If we want to go with the frameless window it might be a matter of creating a custom menu bar. Which I'm happy to look into right now. Other than that the other option is to just have this style on OSX, but for continuity it would look better if we had it on windows as well. The only thing is I don't have a windows machine for a few days to test on, so if someone could when I make the next commit that would be great!

**Comment by StormBurpee** [comment]
Hey @ZeroX-DG and everyone else. I've made it so it works with windows now and has the menu bar. I was still testing this on a mac, hence why it has the extra menu options. But it will adapt to all the menus for the specified operating system. Here are a few images, let me know what you guys think.

**So the mac version looks like this**
screen shot 2018-05-25 at 2 16 29 pm

**And now the windows version will look like this**
screen shot 2018-05-25 at 4 56 03 pm
_You might want to play with the svg icons for min/max/close and make them suit your style a bit better. However I did what I could to make it look neat._

On windows this also works with themes, so it follows along with the same style as mac! Just it's a solid colour this way.

**Themes**
screen shot 2018-05-25 at 4 56 58 pm
screen shot 2018-05-25 at 4 56 27 pm

File: browser/main/Main.js [hunk 1]
```diff
@@ -297,6 +338,39 @@ class Main extends React.Component {
...
         onMouseUp={(e) => this.handleMouseUp(e)}
       >
+      <div className={config.is_win ? 'windows-app-header' : 'mac-app-header'} styleName={config.is_win ? 'windows-app-header' : 'mac-app-header'}>
+        <div className='toolbar'
+          styleName='toolbar'>
+          <div className='toolbar-controls'
+            styleName='toolbar-controls'>
+            <ul className="menubar-list" styleName="menubar-list">
+              {Object.keys(this.state.menu).map(key => {
+                return <li className="menuItem" styleName="menu-item" onBlur={this.closemenu} onClick={() => this.openSubmenu(key)}>{this.state.menu[key].label}<ApplicationMenuDropdown menu={this.state.menu[key].submenu} open={this.state.openmenu == key ? true : false}></ApplicationMenuDropdown></li>;
```
Comment (sosukesuzuki) [inline browser/main/Main.js]: This line is so long! please indention!

Comment (sosukesuzuki) [inline browser/main/Main.js]: please write like below(and fix from eslint):
```jsx
Object.keys(this.state.menu).map(key =>
  <li
    className="menuItem"
    styleName="menu-item"
    onBlur={this.closemenu}
    onClick={() => this.openSubmenu(key)}>
      {this.state.menu[key].label}
    <ApplicationMenuDropdown
      menu={this.state.menu[key].submenu}
      open={this.state.openmenu == key ? true : false} />
  </li>)
```

File: browser/main/Main.js [hunk 2]
```diff
@@ -297,6 +338,39 @@ class Main extends React.Component {
         onMouseMove={(e) => this.handleMouseMove(e)}
         onMouseUp={(e) => this.handleMouseUp(e)}
       >
+      <div className={config.is_win ? 'windows-app-header' : 'mac-app-header'} styleName={config.is_win ? 'windows-app-header' : 'mac-app-header'}>
```
Comment (sosukesuzuki) [inline browser/main/Main.js]: why did you writer both `className` and `styleName`?
I think `className` is unnecessary.

Comment (sosukesuzuki) [reply browser/main/Main.js]: The same applies to other components.

File: browser/main/Detail/ApplicationMenuDropdown.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+  render () {
+    const { className } = this.props
+
+    return (
+      <div className="ApplicationMenuDropdown"
+        styleName={this.state.open == true ? "root" : "root-hidden"}>
+        <ul className="dropdown-items"
+          styleName="dropdown-items">
+          {Object.keys(this.state.menu).map(key => {
```
Comment (sosukesuzuki) [inline browser/main/Detail/ApplicationMenuDropdown.js]: please remove `{}` and `return`.

**Comment by StormBurpee** [comment]
Hi, @sosukesuzuki 
I've made all the changes requested in the review. Let me know if I need to do anything else. 

Regards

**Comment by sosukesuzuki** [comment]
@StormBurpee 
please follow eslint (`/tests/dataApi/createNoteFromUrl-test.js`).
please fix conflict.
(I'll review the test you write later.)

File: browser/main/Detail/ApplicationMenuDropdown.js [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+  constructor (props) {
+    super(props)
+
+    this.state = {
+      isActive: false,
+      menu: props.menu.items,
+      open: props.open
+    }
+
+    console.log(this.state.menu, 'submenu')
```
Comment (sosukesuzuki) [inline browser/main/Detail/ApplicationMenuDropdown.js]: Why console?
I want you to remove `console.log`.

**Comment by Rokt33r** [comment]
@StormBurpee Please remove `package-lock.json`. It generates too many changes. And, could you fix the conflict?

**Comment by Rokt33r** [comment]
Anyway, nice work! 😃 💯

**Comment by sosukesuzuki** [comment]
@StormBurpee please confirm my review!

**Comment by kazup01** [comment]
How is the current state?

**Comment by Rokt33r** [comment]
Close this pr now because it seems to be abandoned. If anyone is interested in this issue, please create another one.

**Comment by mikaoelitiana** [comment]
I have created a new PR from this one and will fix issues from there https://github.com/BoostIO/Boostnote/pull/2318

**Pull request closed without merging**